pkg/compute/serve.go

```diff
@@ -0,0 +1,309 @@
+package compute
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/fastly/cli/pkg/cmd"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	fstexec "github.com/fastly/cli/pkg/exec"
+	"github.com/fastly/cli/pkg/filesystem"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/cli/pkg/update"
+)
+
+// ServeCommand produces and runs an artifact from files on the local disk.
+type ServeCommand struct {
+	cmd.Base
+	manifest         manifest.Data
+	build            *BuildCommand
+	viceroyVersioner update.Versioner
+
+	// Build fields
+	name       cmd.OptionalString
+	lang       cmd.OptionalString
+	includeSrc cmd.OptionalBool
+	force      cmd.OptionalBool
+
+	// Viceroy fields
+	env cmd.OptionalString
+}
+
+// NewServeCommand returns a usable command registered under the parent.
+func NewServeCommand(parent cmd.Registerer, globals *config.Data, build *BuildCommand, viceroyVersioner update.Versioner) *ServeCommand {
+	var c ServeCommand
+
+	c.build = build
+	c.viceroyVersioner = viceroyVersioner
+
+	c.Globals = globals
+	c.CmdClause = parent.Command("serve", "Build and run a Compute@Edge package locally")
+
+	c.manifest.File.SetOutput(c.Globals.Output)
+	c.manifest.File.Read(manifest.Filename)
+
+	// Build flags
+	c.CmdClause.Flag("name", "Package name").Action(c.name.Set).StringVar(&c.name.Value)
+	c.CmdClause.Flag("language", "Language type").Action(c.lang.Set).StringVar(&c.lang.Value)
+	c.CmdClause.Flag("include-source", "Include source code in built package").Action(c.includeSrc.Set).BoolVar(&c.includeSrc.Value)
+	c.CmdClause.Flag("force", "Skip verification steps and force build").Action(c.force.Set).BoolVar(&c.force.Value)
+
+	// Viceroy flags
```

Comment:
To complement my request for a skip building flag above, I think it would be nice if we also allowed users to provide the binary file to run as an optional flag `--file` or `--bin`. This could default to `bin/main.wasm`.

```suggestion
	// Viceroy flags
        c.CmdClause.Flag("file", "The Wasm file to run").Action(c.file.Set).StringVar(&c.file.Value)	
```

Reply:
OK. I'll take a look.